core/src/main/java/ai/timefold/solver/core/impl/heuristic/selector/move/generic/RuinRecreateConstructionHeuristicPhase.java

```diff
@@ -15,24 +13,14 @@ final class RuinRecreateConstructionHeuristicPhase<Solution_>
         super(builder);
     }
 
-    @Override
-    protected void collectMetrics(AbstractStepScope<Solution_> stepScope) {
-        // Nested phase doesn't collect metrics.
-    }
-
     @Override
     protected ConstructionHeuristicPhaseScope<Solution_> buildPhaseScope(SolverScope<Solution_> solverScope, int phaseIndex) {
         return new RuinRecreateConstructionHeuristicPhaseScope<>(solverScope, phaseIndex);
     }
 
     @Override
-    protected void processWorkingSolutionDuringStep(ConstructionHeuristicStepScope<Solution_> stepScope) {
-        // Ruin and Recreate CH doesn't process the working solution, it is a nested phase.
-    }
-
-    @Override
-    protected void updateBestSolutionAndFire(ConstructionHeuristicPhaseScope<Solution_> phaseScope) {
```

Comment:
There wasn't any immediate problem. But since the nested phase shouldn't make any changes to the external best solution either, I decided to include it.